examples/platform_channel_swift/ios/Runner/AppDelegate.swift

```diff
@@ -20,18 +20,21 @@ enum MyFlutterErrorCode {
 }
 
 @main
-@objc class AppDelegate: FlutterAppDelegate, FlutterStreamHandler {
+@objc class AppDelegate: FlutterAppDelegate, FlutterStreamHandler, FlutterPluginRegistrant {
   private var eventSink: FlutterEventSink?
 
   override func application(
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-    GeneratedPluginRegistrant.register(with: self)
-    guard let controller = window?.rootViewController as? FlutterViewController else {
-      fatalError("rootViewController is not type FlutterViewController")
-    }
+    pluginRegistrant = self
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+  }
+
+  func register(with registry: FlutterPluginRegistry) {
+    GeneratedPluginRegistrant.register(with: registry)
+    let registrar = registry.registrar(forPlugin: "battery")
```

Comment:
@vashworth I think I may have misspoken.  I was thinking we could get rid of this since we would be sending the FlutterViewController.  I decided to send the FlutterPluginRegistry though since that makes more sense to send the abstraction that is more apropos.  We could potentially allow people to get the binary messenger from the FlutterPluginRegistry if we wanted to get rid of it.

Reply:
if we have to issue breaking changes, we might as well introduce a `registry.registrar(forChannel:)` for sanity

Reply:
`registry.registrar(forChannel:)` wouldn't quite be the right API since a registrar has a BinaryMessenger which you specify the channel name on.

`Registry.binaryMessenger` would be the convenience we want but that completely circumvents the need for the registrar.

I think `Registry.unaffiliatedRegistrar` would probably be what we want, something like that.

Reply:
Can we get the FlutterViewController and get the binary messenger from it?

Reply:
> Can we get the FlutterViewController and get the binary messenger from it?

If someone has a FlutterViewController they can get the binary messenger and the FlutterViewController is a FlutterPluginRegistry.  There's no concept the represents both of those things.  Having them both doesn't make sense because accessing the binary messenger invalidates the need for a registrar.

Making register callback send in a FlutterViewController would make that possible, but I didn't want to do that because that leaves us tied to a concrete class which gives us less options in the future.